schemas/constructs/component/v1beta1/component.json

```diff
@@ -0,0 +1,131 @@
+				},
+				"genealogy": {
+					"type": "string",
+					"description": "Relationships define the nature of interaction between interconnected components in Meshery. Genealogy represents the various forms of relationships among and between components, like hierarchical (child, parent, grandparent, great-grandparent), interconnected, and dependent, affinal (), collateral (sibling, nieces and nephews, cousin, aunts and uncles), Removed Relationships: relationships among and between components no matter the genealogy of the relationship such as parent, siblings, binding. Relationships have selectors, selector sets, metadata, and optional parameters."
+				},
+				"isAnnotation": {
+					"type": "boolean",
+					"description": "Identifies whether the component should be treated as a deployable entity or just for logical representation."
+				},
+				"shape": {
+					"type": "string",
+					"description": "Shape information for graphical representation."
+				},
+				"primaryColor": {
+					"type": "string",
+					"description": "Primary color for graphical representation."
+				},
+				"secondaryColor": {
+					"type": "string",
+					"description": "Secondary color for graphical representation."
+				},
+				"svgWhite": {
+					"type": "string",
+					"description": "SVG representation in white color."
+				},
+				"svgColor": {
+					"type": "string",
+					"description": "SVG representation in colored form."
+				},
+				"svgComplete": {
+					"type": "string",
+					"description": "Complete SVG representation."
+				},
+				"published": {
+					"type": "boolean",
+					"description": "Controls whether the component should be registered inside the registry."
+				}
+			},
+			"additionalProperties": true
+		},
+		"entity": {
+			"type": "object",
+			"description": "Entity information related to the component.",
+			"properties": {
+				"apiVersion": {
+					"type": "string",
+					"description": "API Version of the object to manage. Example: APIVersion of the Pod object"
+				},
+				"kind": {
+					"type": "string",
+					"description": "Kind of the object to manage. Example: Pod is a Kubernetes object of Kind 'Pod'"
+				},
+				"type": {
+					"description": "Type of the underlying infrastructure the component belongs to. Example: Kubernetes, Terraform."
+				},
+				"subType": {
+					"type": "string",
+					"description": "Used for further classification of the component."
+				},
+				"modelVersion": {
+					"$ref": "https://schemas.meshery.io/core.json#versionString",
+					"description": "Version of the model to which the component belongs."
+				},
+				"schema": {
+					"type": "string",
+					"description": "JSON schema of the object. Example JSON schema of the Pod object"
+				}
+			},
+			"required": [
+				"apiVersion",
+				"kind",
+				"type",
+				"modelVersion",
```

Comment:
Should we remove `modelVersion` as it is already captured inside model property.